addons/sale/wizard/sale_payment_link.py

```diff
@@ -24,6 +24,19 @@ def default_get(self, fields):
             })
         return res
 
+    @api.depends('company_id', 'partner_id', 'currency_id')
+    def _compute_payment_acquiers(self):
+        for link in self:
+            kwargs = {
+                'company_id': self.company_id.id,
+                'partner_id': self.partner_id.id,
+                'currency_id': self.currency_id.id,
+            }
+            if link.res_model == 'sale.order':
+                kwargs['sale_order_id'] = link.res_id
+            link.available_acquirer_ids = link.env['payment.acquirer']._get_compatible_acquirers(
+                **kwargs)
```

Comment:
Leave the management to super as much as possible ;)
(and do not use self in a for x in self loop)
```suggestion
        sale_links = self.filtered(lambda link: link.res_model == 'sale.order')
        super(PaymentLinkWizard, self-sale_links)._compute_payment_acquirers(self)
        for link in sale_links:
            link.available_acquirer_ids = link.env['payment.acquirer']._get_compatible_acquirers(
                company_id=link.company_id.id,
                partner_id=link.partner_id.id,
                currency_id=link.currency_id.id,
                sale_order_id=link.res_id)
```